purchase_order_type/models/res_partner.py

```diff
@@ -7,6 +7,11 @@
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
+    def _default_purchase_type(self):
+        return self.env["purchase.order.type"].search([], limit=1)
+
     purchase_type = fields.Many2one(
-        comodel_name="purchase.order.type", string="Purchase Order Type"
+        comodel_name="purchase.order.type",
+        string="Purchase Order Type",
+        default=lambda self: self._default_purchase_type(),
```

Comment:
This shouldn't have a default.

Reply:
I have made this change because using `Form()` in the test to create a purchase order, the test always fails because there is no `order_type` defined.

Any suggestions on how to remove this change so that the test does not fail?

Reply:
I don't get the problem. Please always put the error.

Reply:
https://app.travis-ci.com/github/OCA/purchase-workflow/jobs/527335286#L1826

Reply:
Sorry, I'm seeing now the problem and indeed although the solution is a bit dirty, we need to add it for compatibility. Please re-introduce it. I don't have any other alternative right now.

Reply:
Looking again the error, the problem was on the purchase order, not on the partner, so I see again that this is not needed. In fact, I mixed both when I checked. There should be something preventing the default in purchase order (probably the computed writable, switching to `False` when changing partner, but should be fixed there.

Reply:
Ok, thanks, changes done (removed `_onchange_company()` to prevent set to `False`).